api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -96,10 +98,19 @@ message DownstreamTlsContext {
 }
 
 // TLS context shared by both client and server TLS contexts.
-// [#next-free-field: 9]
+// [#next-free-field: 11]
 message CommonTlsContext {
   option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.auth.CommonTlsContext";
 
+  // Config for CA client plugin to get secrets
```

Comment:
One of the problems with SDS was the verbose config - if you have 5000 services you'll need to duplicate that message for each cluster and listener.  It would be ideal to not repeat it :-)

There are discussions about better naming and referencing of resources - could this be expressed as just a reference (the name of the provider) ? For most servers there will be a single 'plugin config' - maybe few, but not one per cluster and listener.

Reply:
That's a good point and I have thought about it too but it's not easy to do it in a stateless way. If you have any ideas that do not impose any architectural requirements, let me know.

Reply:
What do others think about creating a new resource type just for this? That way the config is available as a resource and its name can be referenced here. I would prefer not to do it in this PR.

Reply:
Yeah, let's punt this to another PR. Ideally we motivate any additional API decomposition with concrete use cases.